### PR TITLE
Linear gradient

### PR DESCRIPTION
**Pull request by cmaycumber**
Adds a linear gradient component to solve: #73

### CODE REVIEW

**Comment by nandorojo** [comment]
I'm thinking we should move the gradient into its own package so that people who aren't using it don't need to install the extra dependency. This is what I did for Moti's skeleton.

File: packages/gradient/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { LinearGradient } from 'expo-linear-gradient'
+import { styled, useDripsyTheme } from '@dripsy/core'
+
+type Props = Omit<React.ComponentProps<typeof LinearGradient>, 'colors'> & {
+  gradient?: string
+  colors?: string[]
+}
+
+const Grad = styled(
```
Comment (nandorojo) [inline packages/gradient/src/index.tsx:10]: One thing I noticed with the PR: the theme has a linear gradients prop, but there's no consumption of them here. I assume this component should destructure the variant prop and use this to check the theme for a given variant.

Comment (nandorojo) [reply packages/gradient/src/index.tsx:10]: Disregard this, I now see it with the gradients prop.

**Comment by cmaycumber** [comment]
> I'm thinking we should move the gradient into its own package so that people who aren't using it don't need to install the extra dependency. This is what I did for Moti's skeleton.

Everything should be in the gradient package. Let me know if I missed something or maybe accidentally exported it from core. I'll take a look through it again later today.

**Comment by nandorojo** [comment]
Looks like it's exported in the main Dripsy package, but it should require installing `@dripsy/gradient`

**Comment by cmaycumber** [comment]
Ahh I gotcha I'll make the change

**Comment by nandorojo** [comment]
Also, the dependency for Dripsy/core should have a specified version instead of *

File: packages/core/src/css/types.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { LinearGradient } from 'expo-linear-gradient'
```
Comment (nandorojo) [inline packages/core/src/css/types.ts:1]: we should omit this

**Comment by nandorojo** [comment]
We should probably change this to use `expo-module prepare` for the build step too, although I want to see what's up with #86. We might need a custom webpack config for web apps, like this one from moti: https://moti.fyi/web

File: packages/core/src/css/types.ts (hunk 2 above)
Comment (cmaycumber) [reply packages/core/src/css/types.ts]: My latest commit should solve this.

**Comment by cmaycumber** [comment]
> We should probably change this to use `expo-module prepare` for the build step too, although I want to see what's up with #86. We might need a custom webpack config for web apps, like this one from moti: https://moti.fyi/web

Do you think we should add this before merging in or do you still want to experiment a bit more with how expo-module's is affecting the web projects?

**Comment by nandorojo** [comment]
We can leave it with Bob for now to keep comparing, even though the expo module scripts is merged to master in the other packages.

File: examples/example/App.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import {
   DripsyProvider,
   Container,
   Theme,
+  LinearGradient,
```
Comment (nandorojo) [inline examples/example/App.tsx:10]: This should come from the unique package btw

File: packages/gradient/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      const { colors, linearGradients } = useDripsyTheme().theme
+      // Helper function to map colors to appropriate theme colors
+      const colorArrayToTheme = (colorArray: typeof props.colors) => {
+        // Return an empty array if the colors come back as undefined
+        return (
+          colorArray?.map((color) => (colors?.[color] as string) ?? color) ?? []
+        )
+      }
+      let gradientColors: string[] = []
+      if (props.gradient) {
```
Comment (nandorojo) [inline packages/gradient/src/index.tsx:22]: We should maybe add a check / warn a user if `linearGradients?.[props.gradient]` doesn't exist.

**Comment by nandorojo** [comment]
Made some changes, last thing is so fix the merge conflicts and update the example

**Comment by nandorojo** [comment]
https://theme-ui.com/migrating/ migration guide here

**Comment by nandorojo** [comment]
Seems like the biggest breaking change to deal with is the fact that emotion upgrades from 10 -> 11. Besides that, it's the JSX pragma syntax, and then some small stuff.

**Comment by cmaycumber** [comment]
> https://theme-ui.com/migrating/ migration guide here

Sweet this should be helpful. I'll take a look at this tonight.

**Comment by nandorojo** [comment]
One seemingly large breaking change is that the `colors` in the `useThemeUI` hook has changed to `rawColors`... 😬 

I assume this is due to CSS variables? I'm not sure. Maybe we can just override that by reassigning `rawColors: colors` to start.

**Comment by nandorojo** [comment]
Sorry not sure why I put those theme-ui notes on this PR. Got it mixed up.

**Comment by nandorojo** [comment]
This PR is good to go, just need to fix the import in the example, make sure the gradient is **not** exported by the main package, and fix the remaining merge conflicts that GitHub is upset about.

**Comment by cmaycumber** [comment]
> This PR is good to go, just need to fix the import in the example, make sure the gradient is **not** exported by the main package, and fix the remaining merge conflicts that GitHub is upset about.

Will do.

**Comment by cmaycumber** [comment]
@nandorojo This should be gtg. Let me know.

**Comment by nandorojo** [comment]
Cool thanks. Thoughts on exporting it as a named variable rather than default from `dripsy/gradient`?

**Comment by cmaycumber** [comment]
> Cool thanks. Thoughts on exporting it as a named variable rather than default from `dripsy/gradient`?

I'd prefer that. I think that will help keep things consistent.

**Comment by cmaycumber** [comment]
Do you have a preference over just Gradient or LinearGradient?

**Comment by nandorojo** [comment]
Was just thinking that.

LinearGradient is more clear, since that's what it does (no radial, etc). Also consistent with the expo naming. It also sounds lamer, though.

I prefer `Gradient`. I'll sit on it for a few minutes but I say let's go with that. Sounds cleaner.

**Comment by cmaycumber** [comment]
> I prefer Gradient. I'll sit on it for a few minutes but I say let's go with that. Sounds cleaner.

Cool, I agree and it's nice that it's the name as the package name.

**Comment by nandorojo** [comment]
Let's stick with that

**Comment by nandorojo** [comment]
Pointing this to the v2 branch

**Pull request merged**